Extensions/3D/Cube3DRuntimeObjectPixiRenderer.ts

```diff
@@ -75,14 +80,36 @@ namespace gdjs {
     ) {
       const geometry = new THREE.BoxGeometry(1, 1, 1);
       // TODO (3D) - feature: support color instead of texture?
-      const materials = [
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[0]),
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[1]),
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[2]),
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[3]),
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[4]),
-        getFaceMaterial(runtimeObject, materialIndexToFaceIndex[5]),
-      ];
+      const materials: THREE.Material[] = [];
+
+      for (let i = 0; i < 6; i++) {
+        const material: THREE.Material = getFaceMaterial(
+          runtimeObject,
+          materialIndexToFaceIndex[i]
+        );
+        const basicMaterial: THREE.MeshBasicMaterial = new THREE.MeshBasicMaterial();
+        basicMaterial.copy(material);
+        materials.push(
+          basicMaterial.map
+            ? getFaceMaterial(runtimeObject, materialIndexToFaceIndex[i])
+            : new THREE.MeshBasicMaterial({ vertexColors: true })
+        );
```

Comment:
This seems unnecessary to me.
I feel like you could just do: 
```js
const material = ...
materials.push(material)
```
since `getFaceMaterial` already returns a `MeshBasicMaterial` if no resource